src/backend/controllers/battleController.js

```diff
@@ -0,0 +1,246 @@
+// @flow strict
+
+const { validationResult } = require('express-validator');
+
+import type { Request, Response } from 'express';
+
+// npm library for elo calculations
+const EloRating = require('elo-rating');
+
+const User = require('../../models/userModel');
+const BattleRecord = require('../../models/battleRecordModel');
+const Tank = require('../../models/tankModel');
+
+exports.prepareMatch = async (req: Request, res: Response) => {
+
+    // Contains errors for failed validation.
+   const errors = validationResult(req);
+
+   if (!errors.isEmpty()) {
+       // Return 400 for a bad request
+       return res
+           .status(400)
+           .json({ errors: errors.array() });
+   }
+   const { personBeingChallengedId, challengerTankId } = req.body;
+
+   try{
+    // Need to do query explicityly so that the returned doc isn't local to the try catch
+    const personBeingChallengedUserDoc = await User.findById(personBeingChallengedId, 'favoriteTank wager money stats.elo');
+
+    if(personBeingChallengedUserDoc == null){
+        return res
+            .status(404)
+            .json({ msg: 'Could not find the personBeingChallenged in DB'})
+    }
+
+    if(personBeingChallengedUserDoc.money < personBeingChallengedUserDoc.wager){
+        return res
+            .status(401)
+            .json({ msg: 'personBeingChallenged does not have enough money to wager'});
+    }
+
+    const personBeingChallengedTank = await Tank.findById(personBeingChallengedUserDoc.favoriteTank);
+
+    if(personBeingChallengedTank == null){
+        return res
+            .status(404)
+            .json({ msg: "Could not find the personBeingChallenged's tank in DB"});
+    }
+
+    const challengerTank = await Tank.findById(challengerTankId);
+
+    if(challengerTank == null){
+        return res
+            .status(404)
+            .json({ msg: "Could not find the challenger's Tank in DB"});
+    }
+
+    // Query the amount of money the challenger has
+    const challengerUserDoc = await User.findById(challengerTank.userId, 'money');
+
+    if(challengerUserDoc == null){
+        return res
+            .status(404)
+            .json({ msg: 'Could not find the challenger in DB'});
+    }
+
+    if(challengerUserDoc.money < personBeingChallengedUserDoc.wager){
+        return res
+            .status(401)
+            .json({ msg: 'Challenger does not have enough money to wager'});
+    }
+
+    const newRecord = new BattleRecord({
+        userOne: personBeingChallengedId,
+        userTwo: challengerTank.userId,
+        tankOne: personBeingChallengedTank,
+        tankTwo: challengerTank,
+        winner: -1,
+        prizeMoney: personBeingChallengedUserDoc.wager
```

Comment:
Just want to be certain, wasn't the prize money essentially the sum of both players wagers?

Reply:
ah yess brain fart thanks for the catch